uast/transformer/semantic.go

```diff
@@ -161,96 +164,102 @@ func (c *commentElems) isTab(r rune) bool {
 	return false
 }
 
-func (c *commentElems) Split(text string) bool {
-	if c.DoTrim {
-		text = strings.TrimLeftFunc(text, unicode.IsSpace)
+func max(x, y int) int {
+    if x > y {
+        return x
+    }
+    return y
+}
+
+// Returns the first index i of runes satisfying f(runes[i]), len(runes) otherwise
+func firstIndexFunc(runes []rune, f func(r rune) bool) int {
+	for i, r := range runes {
+		if (f(r)) {
+			return i;
+		}
 	}
+	return len(runes);
+}
 
-	if !strings.HasPrefix(text, c.StartToken) || !strings.HasSuffix(text, c.EndToken) {
-		return false
+// Returns the last index i of runes satisfying f(runes[i]), -1 otherwise
+func lastIndexFunc(runes []rune, f func(r rune) bool) int {
+	for i := range runes {
+		j := len(runes) - 1 - i
+		if (f(runes[j])) {
+			return j;
+		}
 	}
-	text = strings.TrimPrefix(text, c.StartToken)
-	text = strings.TrimSuffix(text, c.EndToken)
+	return -1;
+}
 
-	// find prefix (if not already trimmed)
-	i := strings.IndexFunc(text, func(r rune) bool {
-		return !c.isTab(r)
-	})
+func (c *commentElems) findPrefix(f func(r rune) bool) {
+	runes := []rune(c.Text)
+	i := firstIndexFunc(runes, f)
```

Comment:
so maybe these functions should return a substring instead of index

Reply:
You mean `firstIndexFunc` and `lastIndexFunc`?

Reply:
both

Reply:
The downside of that is that I would have to call `TrimPrefix` or `TrimSuffix` inside `findPrefix` or `findSuffix`. And `firstIndexFunc` is also used to compute only the prefix (as in the tab calculation inside the loop), so making them return `(prefix, rest of the text)` or `(text minus suffix, suffix)` is not suitable either. Also it should not return strings IMHO because if it is used to compute the tab, therefore it needs runes. If it returns strings I am adding another conversion `string <-> [] rune` again